python/ray/serve/tests/test_proxy_state.py

```diff
@@ -23,6 +23,17 @@
 HEAD_NODE_ID = "node_id-index-head"
 
 
+class FakeTime:
```

Comment:
consider putting this in `tests/utils` so that it can be reused :)

Reply:
along w/ an ABC would be nice as suggested above

Reply:
will do!

Reply:
Interesting, there is already this [`MockTimer`](https://github.com/ray-project/ray/blob/master/python/ray/serve/tests/common/utils.py#L22) so just gonna reuse it XP